src/ln/channelmanager.rs

```diff
@@ -1439,6 +1439,31 @@ impl ChannelManager {
 		Ok(accept_msg)
 	}
 
+	fn internal_accept_channel(&self, their_node_id: &PublicKey, msg: &msgs::AcceptChannel) -> Result<(), MsgHandleErrInternal> {
+		let (value, output_script, user_id) = {
+			let mut channel_state = self.channel_state.lock().unwrap();
+			match channel_state.by_id.get_mut(&msg.temporary_channel_id) {
+				Some(chan) => {
+					if chan.get_their_node_id() != *their_node_id {
+						//TODO: see issue #153 , need a consistent behavior on obnoxious behavior from random node
+						return Err(MsgHandleErrInternal::from_no_close(HandleError{err: "Got a message for a channel from the wrong node!", action: Some(msgs::ErrorAction::IgnoreError)}));
```

Comment:
Oops, well it needs to be consistent but either between SendErrorMessage or Disconnect with an ErrorMessage specified. IgnoreError isnt really correct either way.